client/src/main/java/io/pravega/client/byteStream/impl/ByteStreamWriterImpl.java

```diff
@@ -60,6 +73,12 @@ public void flush() throws IOException {
         out.flush();
     }
 
+    @Override
+    @Synchronized
+    public CompletableFuture<Void> flushAsync() {
+        return this.latestEventFuture;
```

Comment:
Above flush is calling `out.flush` here no method is called. It would be good if there were a method defined on the output stream to notify it of the fact that write flush async was requested even if it does nothing. This will ensure that when we make changes to the segmentOutputStreamImpl that we think about this code path.
In practice we can decide if we want to trigger a keepAlive in response to that method. As we discussed that is a tradeoff. For now I think adding an empty method and a comment mentioning it is sufficient.

Reply:
Added an empty method in OutputStream as specified.